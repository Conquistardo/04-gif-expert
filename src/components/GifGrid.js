import React from 'react'
import { useFetchGift } from '../hooks/useFetchGift';
import { GifGridItems } from './GifGridItems';
import PropTypes from 'prop-types';

export const GifGrid = ({categoria}) => {

    const {data:images, loading} = useFetchGift(categoria);

    return (
        <>
            <h3>{categoria}</h3>
            {loading?<p>Cargando...</p>:''}
            <div className='cardGrid'>            
                    {
                        images.map(
                            (img) => <GifGridItems 
                                    key={img.id}
                                    {...img}/>
                        )
                    }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired
}
