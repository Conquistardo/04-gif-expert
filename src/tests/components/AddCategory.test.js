import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas del componente AddCategory', () => {

    const setCategorias = jest.fn();
    let wrapper = shallow(<AddCategory setCatergorias={setCategorias}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCatergorias={setCategorias}/>);
    });

    test('debe de mostrarse', () => {        
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Prueba';
        input.simulate('change', { target: {value: value} });        
    });

    test('no debe de postear la informaciÃ³n on submit', () => {
        const form = wrapper.find('form');
        form.simulate('submit', {
            preventDefault: () => {}
        });

        expect(setCategorias).not.toHaveBeenCalled();

    });

    test('debe de llamar el setCategorias y limpiar la caja de texto', () => {
        const value = 'Valor';
        let input = wrapper.find('input');
        input.simulate('change', { target: {value: value} });    
        const form = wrapper.find('form');
        form.simulate('submit', {
            preventDefault: () => {}
        });

        expect(setCategorias).toHaveBeenCalledWith(expect.any(Function));
        expect(setCategorias).toHaveBeenCalled();

        input = wrapper.find('input');
        const valorInput = input.props().value;

        expect(valorInput).toBe('');
        
    });
    
    
});