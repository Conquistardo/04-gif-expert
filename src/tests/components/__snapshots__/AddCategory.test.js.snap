// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente AddCategory debe de mostrarse 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
